code/shared/os/Firewall.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+#ifdef _WIN32
+#include <iwscapi.h>
+#include <wscapi.h>
+#include <versionhelpers.h>
+#endif
+
+namespace os
+{
+std::wstring GetActiveFirewallName(std::wstring defaultName = L"Unknown")
+{
+#ifdef _WIN32_WINNT >= 0x0602 // Need at least windows 8 to compile this
+
+	// IWSCProductList and IWscProduct are Windows 8+ Desktop
+	if (IsWindows8OrGreater())
+	{
+		CoInitializeEx(0, COINIT_APARTMENTTHREADED);
+
+		IWSCProductList* productList = nullptr;
```

Comment:
`Microsoft::WRL::ComPtr<T>` from `wrl.h` as in other places might be useful instead.